From 74812daa4a55fa57ea86576592e7687204e20538 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Sun, 26 Jun 2022 03:47:06 +0200
Subject: [PATCH 07/54] ODROID-HC4: add DT attributes to enable fan1_input

- from vendor kernel modified DT
- this allows userspace fancontrol/pwmconfig to work
---
 arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
index e3486f60645a..89ef6f07603c 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
@@ -23,6 +23,10 @@ fan0: pwm-fan {
 		cooling-max-state = <3>;
 		cooling-levels = <0 120 170 220>;
 		pwms = <&pwm_cd 1 40000 0>;
+		fan-supply = <&vcc_5v>;
+		interrupt-parent = <&gpio_intc>;
+		interrupts = <84 IRQ_TYPE_EDGE_FALLING>;
+		pulses-per-revolutions = <2>;
 	};
 
 	leds {
-- 
2.30.2

